tests/django_pg_migration_tools/test_operations.py

```diff
@@ -427,3 +454,355 @@ def test_when_not_allowed_to_migrate(self):
+                operation.database_forwards(
+                    self.app_label, editor, project_state, new_state
+                )
+
+        # Drop the constraint. We aren't in a test with transaction, we have
+        # to clean up.
+        with connection.cursor() as cursor:
+            cursor.execute(_DROP_CONSTRAINT_QUERY)
+
+    @pytest.mark.django_db(transaction=True)
+    def test_do_nothing_when_asked_not_to_raise_when_constraint_exists(self):
+        project_state = ProjectState()
+        project_state.add_model(ModelState.from_model(IntModel))
+        new_state = project_state.clone()
+
+        # Create the constraint. The operation won't raise an error when the
+        # constraint already exists because `raise_if_exists` is False.
+        with connection.cursor() as cursor:
+            cursor.execute(_CREATE_CONSTRAINT_QUERY)
+
+        operation = operations.SaferAddUniqueConstraint(
+            raise_if_exists=False,
+            model_name="intmodel",
+            constraint=UniqueConstraint(
+                fields=("int_field",),
+                name="unique_int_field",
+            ),
+        )
+
+        with connection.schema_editor(atomic=False, collect_sql=False) as editor:
+            with utils.CaptureQueriesContext(connection) as queries:
+                operation.database_forwards(
+                    self.app_label, editor, project_state, new_state
+                )
+        assert len(queries) == 1
+
+        # Only fired one query to check if the index already exists.
+        assert queries[0]["sql"] == dedent("""
+            SELECT conname
+            FROM pg_catalog.pg_constraint
+            WHERE conname = 'unique_int_field';
+            """)
+
+        # Drop the constraint. As we aren't in a test with transaction, we have
+        # to clean up.
+        with connection.cursor() as cursor:
+            cursor.execute(_DROP_CONSTRAINT_QUERY)
+
+    def test_when_not_unique_constraint(self):
+        project_state = ProjectState()
+        project_state.add_model(ModelState.from_model(IntModel))
+
+        with pytest.raises(ValueError):
+            operations.SaferAddUniqueConstraint(
+                model_name="intmodel",
+                # This isn't a valid class!
+                constraint=CheckConstraint(  # type: ignore[arg-type]
+                    name="test_check_constraint",
+                    condition=Q(),  # type: ignore[call-arg]
+                ),
+            )
+
+    def test_when_name_too_long(self):
+        project_state = ProjectState()
+        project_state.add_model(ModelState.from_model(IntModel))
+
+        with pytest.raises(ValueError):
+            operations.SaferAddUniqueConstraint(
+                model_name="intmodel",
+                constraint=UniqueConstraint(
+                    fields=("int_field",),
+                    name="a" * (operations.MAX_RELATION_NAME_LEN + 1),
```

Comment:
In reality, as noted above, this should fail if the name is `operations.MAX_RELATION_NAME_LEN - 3` characters long.

Reply:
Addressed via: https://github.com/kraken-tech/django-pg-migration-tools/pull/26#discussion_r1790934806